Add op_amax support

cccclai opened the pull request:
Summary: As title, add op_amax to support an internal model, add unit test in test_qnn_delegate.py

Differential Revision: D72613814

shewu-quic reviewed (approved):
Thanks for your effort.

There is one small thing need to add.
```
diff --git a/backends/qualcomm/_passes/layout_transform.py b/backends/qualcomm/_passes/layout_transform.py
index 17960a602..4d47c38bc 100644
--- a/backends/qualcomm/_passes/layout_transform.py
+++ b/backends/qualcomm/_passes/layout_transform.py
@@ -47,6 +47,7 @@ class LayoutTransform(ExportPass):
     layout_agnostic_ops = {
         exir_ops.edge.aten.abs.default,
         exir_ops.edge.aten.add.Tensor,
+        exir_ops.edge.aten.amax.default,
         exir_ops.edge.aten.bitwise_or.Tensor,
         exir_ops.edge.aten.bmm.default,
         exir_ops.edge.aten.bitwise_and.Tensor,
```

cccclai commented:
> Thanks for your effort.
> 
> There is one small thing need to add.
> 
> ```
> diff --git a/backends/qualcomm/_passes/layout_transform.py b/backends/qualcomm/_passes/layout_transform.py
> index 17960a602..4d47c38bc 100644
> --- a/backends/qualcomm/_passes/layout_transform.py
> +++ b/backends/qualcomm/_passes/layout_transform.py
> @@ -47,6 +47,7 @@ class LayoutTransform(ExportPass):
>      layout_agnostic_ops = {
>          exir_ops.edge.aten.abs.default,
>          exir_ops.edge.aten.add.Tensor,
> +        exir_ops.edge.aten.amax.default,
>          exir_ops.edge.aten.bitwise_or.Tensor,
>          exir_ops.edge.aten.bmm.default,
>          exir_ops.edge.aten.bitwise_and.Tensor,
> ```

thanks! updated

cccclai committed:
Add op_amax support (#9955)

Summary:

As title, add op_amax to support an internal model, add unit test in test_qnn_delegate.py

Reviewed By: kirklandsign

Differential Revision: D72613814

cccclai commented:
Quick question, what is the reason that [`insert_permute` defined in LayoutTransform](https://github.com/pytorch/executorch/blob/7e8acb9471f52ed7dd5a6f7f11c59514fa048567/backends/qualcomm/_passes/layout_transform.py#L119) defaults to False? I currently manually set it to True to continue lowering, but it seems required anyway, correct?

haowhsu-quic commented:
> Quick question, what is the reason that [`insert_permute` defined in LayoutTransform](https://github.com/pytorch/executorch/blob/7e8acb9471f52ed7dd5a6f7f11c59514fa048567/backends/qualcomm/_passes/layout_transform.py#L119) defaults to False? I currently manually set it to True to continue lowering, but it seems required anyway, correct?

This will actually insert the permute node according to layout information but will break graph if you're currently in partitioner stage (the new layout will not match pytorch cpu convention).
We only turn `insert_permute` to true in `preprocess` stage for QNN to build graph correctly.
